### PR TITLE
#72 Additional columns to result table

### PR DESCRIPTION
**Pull request by scape76**
---
title: Issue #72 
---

## What type of PR is this? (select all that apply)

- [ ] 🍕 Feature

I would say its a task
## Related Tickets & Documents

- Related Issue #
- Closes #

## Added/updated tests?

- [ ] 🙅 no, because they aren't needed

## [optional] Are there any post deployment tasks we need to perform?
npx prisma db push

### CODE REVIEW

File: src/app/race/actions.ts [hunk 1]
```diff
@@ -1,15 +1,21 @@
 "use server";
 
-import { User } from "@prisma/client";
+import { User, Prisma } from "@prisma/client";
 import { prisma } from "@/lib/prisma";
 
 export async function saveUserResult(input: {
   userId: User["id"];
   timeTaken: string | number;
+  errors: number;
```
Comment (shardel23) [inline src/app/race/actions.ts:9]: Consider renaming to errorCount

File: src/app/race/actions.ts [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+  accuracy: number;
 }) {
   if (
-    typeof input.timeTaken !== "string" &&
-    typeof input.timeTaken !== "number"
+    (typeof input.timeTaken !== "string" &&
+      typeof input.timeTaken !== "number") ||
+    typeof input.errors !== "number" ||
+    typeof input.wpm !== "number" ||
+    typeof input.accuracy !== "number"
```
Comment (shardel23) [inline src/app/race/actions.ts:18]: Why do we need to check dynamically the types? Doesn't typescript ensures the types for us?

Comment (scape76) [reply src/app/race/actions.ts:18]: Yes, typescript really does. I saw this practice in some popular repo, but I cant tell the reason for that

File: prisma/schema.prisma [hunk 3]
```diff
@@ -55,8 +55,12 @@ model VerificationToken {
 }
 
 model Result {
-  id        String @id @default(cuid())
-  takenTime String
-  userId    String
-  user      User   @relation(fields: [userId], references: [id], onDelete: Cascade)
+  id         String   @id @default(cuid())
+  takenTime  String
+  finishedAt DateTime @default(now())
```
Comment (webdevcody) [inline prisma/schema.prisma:60]: please merge in main and use new convention for @map on names that have two+ words in them.

File: src/app/race/actions.ts (hunk 2 above)
Comment (scape76) [reply src/app/race/actions.ts:18]: I just asked the author. I will remove this type checking next time, he was just using route handler

**Pull request merged**